_posts/2025-08-07-Diving-into-services-with-OpenSearch-and-OpenTelemetry.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: post
+title: "Diving deep into services with OpenSearch and OpenTelemetry"
+category: blog
+tags: [observability, traces, services, opentelemetry, opensearch-3-1]
+authors:
+    - pshenoy
+    - tackadam
+date: 2025-08-07
+categories:
+  - technical-posts
+meta_keywords: observability, traces, services, OpenTelemetry, OpenSearch 3.1, Investigation, 
+meta_description: Learn how to use OpenSearch 3.1 and OpenTelemetry for end-to-end observability, troubleshooting, and root cause analysis in distributed microservices. This blog demonstrates a practical workflow for identifying, investigating, and resolving service issues using advanced tracing, service maps, and AI-powered features in OpenSearch Dashboards.
+---
+
+In modern distributed systems, understanding the interactions between microservices is crucial for identifying performance bottlenecks and diagnosing failures. In this blog, we'll demonstrate the new features introduced in Trace Analytics plugin in OpenSearch version 3.1, such as enhanced service map visualizations, advanced span grouping, and latency distribution charts alongside OpenTelemetry to collect and instrument traces. We'll then show how to use OpenSearch and OpenSearch Dashboards to explore, visualize, and analyze an Observability investigation workflow. 
+
+## Setting up the demo
+
+We'll use the OpenTelemetry demo for the Astronomy Shop, an e-commerce site composed of multiple microservices. This demo is available in [OpenTelemetry demo](https://github.com/opensearch-project/opentelemetry-demo). A Docker Compose setup launches the Astronomy Shop services and the following components:
+
+* **Data Prepper**: Ingests logs and traces into OpenSearch.
+* **OpenSearch**: Stores telemetry data and serves as the search engine.
+* **OpenSearch Dashboards**: Provides a unified UI for logs, metrics, and traces.
+
+The architecture of the demo is shown in the following diagram.
+
+![OpenSearch Observability with OpenTelemetry](/assets/media/blog-images/2025-08-07-Diving-into-services-with-OpenSearch-and-OpenTelemetry/os-observability-architecture.png){:class="img-centered"}
+
+### OpenTelemetry demo and Astronomy Shop
+
+The Astronomy Shop includes services such as Frontend, Cart, Ad, Accounting, Currency, Payment, and Checkout, as shown in the following image. Each service is implemented in a different language and uses the OpenTelemetry SDK to emit spans and metrics.
+
```

Comment:
Line 32: Isn't it named OpenTelemetry Astronomy Shop?

Reply:
updated here https://github.com/opensearch-project/project-website/pull/3887/commits/322c1c20d2b715af51e1b11ea3ae5157ec436766